picard/ui/options/plugins.py

```diff
@@ -341,11 +344,18 @@ def download_handler(self, response, reply, error, plugin):
             log.error("Error occurred while trying to download the plugin: '%s'" % plugin.module_name)
             return
 
-        self.tagger.pluginmanager.install_plugin(
-            None,
-            plugin_name=plugin.module_name,
-            plugin_data=response
-        )
+        if action == "install":
+            self.tagger.pluginmanager.install_plugin(
```

Comment:
Since those share same arguments, you may do:

```python
if condition:
  func = self.tagger.pluginmanager.install_plugin
else:
  func = self.tagger.pluginmanager.update_plugin

func(None, plugin_name=plugin.module_name, plugin_data=response)
```